page-wrapper.js

```diff
@@ -1,8 +1,8 @@
 import React from "react";
 import Layout from "./src/components/layout.js";
 
-export const wrapPageElement = ({ element }) => (
-  <Layout>
+export const wrapPageElement = ({ element ,props }) => (
```

Comment:
Hi @8rxn,

Nice work! I will be adding some notes to your code.

I believe it is unnecessary to make any changes to this file and `wrapPageElement`, because [passing `element` as `children`](https://react.dev/learn/passing-props-to-a-component#passing-jsx-as-children)  to the `<Layout>` component will give access to `props.location.pathname`.

<details><summary>when you console.log `element`</summary>
element {
  '$$typeof': Symbol(react.element),
  type: [Function: IndexPage],
  key: null,
  ref: null,
  props: {
    path: '/*',
    '*': '',
    uri: '/',
    location: { pathname: '/', search: '', hash: '' },
    navigate: [Function: navigate],
    children: undefined,
    pageContext: {},
    serverData: undefined,
    params: {}
  },
  _owner: null,
  _store: {}
}
children {
  '$$typeof': Symbol(react.element),
  type: [Function: IndexPage],
  key: null,
  ref: null,
  props: {
    path: '/*',
    '*': '',
    uri: '/',
    location: { pathname: '/', search: '', hash: '' },
    navigate: [Function: navigate],
    children: undefined,
    pageContext: {},
    serverData: undefined,
    params: {}
  },
  _owner: null,
  _store: {}
}
</details>

Looking at the console.log for `element`, it has `props`, a nested object which can be destructured from the `children` prop in `layout.js` to access `pathname`. 

Let me know if you have any questions.

Thanks,
Randy

Reply:
Ah. Will implement this change.